File: sum_prime_under_2M.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int limit = 2000000;
    // 소수 여부를 체크할 배열 (true로 초기화)
    vector<bool> is_prime(limit + 1, true);
    
    is_prime[0] = is_prime[1] = false; // 0과 1은 소수가 아님

    // 에라토스테네스의 체 알고리즘
    for (int p = 2; p * p <= limit; p++) {
        // p가 소수라면 p의 배수들을 모두 false로 변경
        if (is_prime[p]) {
            for (int i = p * p; i <= limit; i += p)
                is_prime[i] = false;
        }
    }

    long long sum = 0; // 합계는 큰 범위인 long long 사용
    for (int p = 2; p <= limit; p++) {
        if (is_prime[p]) {
            sum += p;
        }
    }

    cout << "2,000,000 이하 소수의 합: " << sum << endl;

    return 0;
}

File: sum_prime_under_2M.py
import time

def sieve_of_eratosthenes(limit):
    primes = [True] * (limit + 1)
    primes[0] = primes[1] = False

    for number in range(2, int(limit**0.5) + 1):
        if primes[number]:
            for multiple in range(number * number, limit + 1, number):
                primes[multiple] = False

    return [i for i, is_prime in enumerate(primes) if is_prime]

# 1. 측정 시작 시간 기록
start_time = time.time()

limit = 2000000
primes = sieve_of_eratosthenes(limit)
result = sum(primes)

# 2. 측정 종료 시간 기록
end_time = time.time()

# 3. 결과 출력
print(f"2,000,000 이하 소수의 합: {result}")
print(f"수행 시간: {(end_time - start_time) * 1000:.2f} ms") # 밀리초 단위로 변환